javascript/javascript2/week3/fastestPresser/main.js

```diff
@@ -0,0 +1,101 @@
+// 1. Create an input and a button in html. When the button is clicked, get the value of the input.
+//This value will be the amount of time the game should run.
+// 2. Set a timeout for the time specified by the user.
+//After that time has run out just log out a simple string.
+// 3. Create an event listener so you can call a function when any key is pressed.
+//Now grap the actual key that was pressed.Fx was it a j or an i.We are interested in s and l.
+//Here google is your friend!
+// 4. Keep a counter for how many times l and s was pressed.
+// 5. Now put it all together! After the timeout is done figure out which of the counters is largest.
+const topContainerRef = document.getElementById("topContainer");
+const inputDurationRef = document.getElementById("inputDuration");
+const startBtnRef = document.getElementById("startBtn");
+const leftPlayerCounterOutputRef = document.getElementById(
+  "leftPlayerCounterOutput"
+);
+const rightPlayerCounterOutputRef = document.getElementById(
+  "rightPlayerCounterOutput"
+);
+const leftPlayerInterfaceRef = document.getElementById("leftPlayerInterface");
+const rightPlayerInterfaceRef = document.getElementById("rightPlayerInterface");
+const winNotificationRef = document.createElement("p");
+const confettiCanvas = document.createElement("canvas");
+
+let gameDuration = 0;
+let leftPlayerCounter = 0;
+let rightPlayerCounter = 0;
+
+inputDurationRef.addEventListener("input", onInputChange);
+startBtnRef.addEventListener("click", onStartBtnClick);
+
+function onInputChange(e) {
+  gameDuration = parseInt(e.target.value);
+  if (!isNaN(gameDuration)) {
+    startBtnRef.disabled = false;
+    return gameDuration;
+  }
```

Comment:
```suggestion
  if (isNaN(gameDuration)) {
    return;
  }
  
  startBtnRef.disabled = false;
```

Using negation in conditionals makes code harder to read, you can instead flip it and use it as a guard statement. If the condition is not met, the code continues to execute and enables the button.